up-l1/zenoh.adoc

```diff
@@ -35,9 +35,141 @@ Zero Overhead Network Protocol.
 
-
-for more information, please visit https://projects.eclipse.org/projects/iot.zenoh or https://zenoh.io/
+For more information, please visit https://projects.eclipse.org/projects/iot.zenoh or https://zenoh.io/
 
 == Specifications
 
-_Coming Soon_
\ No newline at end of file
+=== Zenoh Config
+
+While constructing the UPClientZenoh object, the Zenoh config needs to be assigned.
+
+=== UAttribute Mapping
+
+Zenoh supports user attachment.
+We can send additional information with the mechanism, for example, UAttribute.
+This can reduce the unnecessary serialization on payload, which mostly takes time.
+
+User attachment supports key-value pairs, and the following table is the key-value pairs using now.
+
+[cols="1,1"]
+|===
+| key | value
+
+| "uattr"
+| UAttribute object encoded into protobuf
+|===
+
+Besides the user attachment, some features in the UAttribute can be mapped into Zenoh transport settings.
+
+==== Message Type
+
+There are 3 kinds of message types in uProtocol (publish, request, response).
+Different Zenoh operations should use the corresponding message types.
+
+[cols="1,1"]
+|===
+| uProtocol message type | Zenoh operations
+
+| publish | put
+| request | get (query)
+| response | reply (queryable)
+|===
+
+==== Priority Mapping (uProtocol to Zenoh):
+
+[cols="1,1"]
+|===
+| uProtocol priority | Zenoh Priority
+
+| CS0 | BACKGROUND
+| CS1 | DATA_LOW
+| CS2 | DATA
+| CS3 | DATA_HIGH
+| CS4 | INTERACTIVE_LOW
+| CS5 | INTERACTIVE_HIGH
+| CS6 | REAL_TIME
+|===
+
+=== URI Mapping
+
+UUri should be able to be serialized into link:../basics/uri.adoc#42-micro-uris[micro form].
+Then we can transform the micro-form UUri, which should be `u8` array, into string in lowercase hexadecimal representation.
+
+Zenoh key for uProtocol consists of three parts and they are separated by `/`.
+Zenoh key **MUST** start with `up/`.
+If UAuthority is defined in UUri, it should be put in the second part.
+Then the rest of the UUri can be put in the last part.
+
+To be more specific:
+
+* UUri without UAthourity: `up/[byte representation of the UUri]`
+* UUri with UAthourity: `up/[byte representation of the UAuthority]/[byte representation of the rest of the UUri]`
+
+Note that there is a special case. If the UUri only contains UAthority with empty UEntity and UResource,
+it's defined as a special form of Zenoh key `up/[byte representation of UAuthority]/**`.
+This is only used by uStreamer.
+
+Some examples:
+
+* UUri without UAthourity:
+
+[source]
+----
+UUri {
+  authority: empty,
+  entity: UEntity {
+    name: "body.access",
+    version_major: 1,
+    id: 1234,
+  }
+  resource: UResource {
+    name: "door",
+    instance: "front_left",
+    message: "Door",
+    id: 5678,
+  }
+}
+----
+
+There is no UAuthority, so we can ignore the second part.
```

Comment:
The complete Zenoh key is `up/<uauthoriy>/<uentity+uresource>`
Since there is no UAuthority in the local URI, we only have `up/<uentity+uresource>`.
That's what I mean ignoring the second part, which is `<uauthority>`